draft-ietf-quic-http.md

```diff
@@ -472,8 +472,8 @@ table. While these updates are not directly part of the message exchange, they
 must be received and processed before the message can be consumed.  See
 {{header-formatting}} for more details.
 
-The "chunked" transfer encoding defined in Section 7.1 of {{?HTTP11}} MUST NOT
-be used.
+Transfer codings (see Section 6.1 of {{?HTTP11}}) are not defined for HTTP/3;
+the Transfer-Encoding header field MUST NOT be used.
```

Comment:
I'm not sure if this MUST NOT aligns well with the existing requirements in [section 4.1.1](https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-field-formatting-and-compre) that states _endpoints MUST NOT generate_, _intermediaries SHOULD remove_.

If the intended requirements are:
* endpoints MUST NOT generate requests that contain connection-specific header fields (e.g., transfer-encoding)
* intermediaries MUST NOT forward chunked encoding without decoding it and SHOULD remove connection-specific header fields
then I think it might be better to explicitly state as such, possibly in section 4.1.1 altogether.

Reply:
That's an interesting point.  I think that's an instance of a non-deterministic SHOULD, rather than an optional behavior per se.  In other words, to make that a MUST, we'd have to have a pointer to the complete list of headers which MUST be removed, and since headers are an extension point, there's no such document.  Someone could define a new hop-by-hop header for HTTP/1.1 and all implementations would instantly be violating this requirement.

In other words, this SHOULD ought to be read as "intermediaries MUST remove all the headers known to be connection-specific, but an intermediary might not know the meaning of every header generated by the endpoint."  If there's better language for that section, I'd welcome a PR with clearer text.

I'm okay with the language for Transfer-Encoding being stronger than the SHOULD which covers the class of hop-by-hop headers; we already have something of an instance of this with Upgrade; it's also covered by the SHOULD in 4.1.1 but 4.3 says it's not supported.  (Perhaps 4.3 should have a MUST about the Upgrade header and the 101 status code...?)